pmd-java/src/main/java/net/sourceforge/pmd/lang/java/rule/bestpractices/PreserveStackTraceRule.java

```diff
@@ -140,13 +141,34 @@ private boolean checkForTargetUsage(String target, Node baseNode) {
             List<ASTName> nameNodes = baseNode.findDescendantsOfType(ASTName.class);
             for (ASTName nameNode : nameNodes) {
                 if (target.equals(nameNode.getImage())) {
-                    match = true;
-                    break;
+                    boolean isPartOfStringConcatenation = isStringConcat(nameNode, baseNode);
+                    if (!isPartOfStringConcatenation) {
+                        match = true;
+                        break;
+                    }
                 }
             }
         }
         return match;
     }
+    
+    /**
+     * Checks whether the given childNode is part of an additive expression (String concatenation) limiting search to base Node.
+     * @param childNode
+     * @param baseNode
+     * @return
+     */
+    private boolean isStringConcat(Node childNode, Node baseNode) {
+        Node currentNode = childNode;
+        // limit to 10 levels
+        for (int i = 0; i < 10 && currentNode != null && currentNode != baseNode; i++) {
```

Comment:
this is rather arbitrary... any reason not to use `getFirstParentOfType`?

Reply:
it's true, that was a (very) defensive measure against infinite loop, but a while checking for base node will be enought.
I've commited the change.